axes/backends.py

```diff
@@ -21,6 +21,9 @@ def authenticate(self, request, username=None, password=None, **kwargs):
         :return: Nothing, but will update return_context with lockout message if user is locked out.
         """
 
+        if request is None:
```

Comment:
@camilonova @aleksihakli addressed pr concerns, but i have another query as to handle when request is None.

Django Rest Framework (which is very widely used) does not set `request` when passing it to authenticate, so things like @authentication_classes((BasicAuthentication,)) will error out. 

options:

- Error out like this
- no-op if no request and log a warning

Reply:
I think this behaviour is fine for the time being. If someone wants to use this custom backend in their project they need to make sure it can function correctly.